src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp

```diff
@@ -187,25 +188,36 @@ namespace OpenMS
       try
       {
         mzid_parser_.parse(mzid_file.c_str());
+      }
+      catch (xercesc::XMLException& e)
+      {
+        char* message = xercesc::XMLString::transcode(e.getMessage());
+        //          ostringstream errBuf;
+        //          errBuf << "Error parsing file: " << message << flush;
+        OPENMS_LOG_ERROR << "XERCES error parsing file: " << message << flush << endl;
+        XMLString::release(&message);
+      }
 
-        // no need to free this pointer - owned by the parent parser object
-        xercesc::DOMDocument* xmlDoc = mzid_parser_.getDocument();
+      // we adopt/own the document so we can free it in case this DOMHandler is reused on another file.
+      xercesc::DOMDocument* xmlDoc = mzid_parser_.adoptDocument();
```

Comment:
does this imply any performance / memory hit?

Reply:
AFAIK not. It is only a difference in ownership of the memory. Either the parser has it, or the class that calls it.